src/features/views/components/ViewColumnDialog/PersonTagConfig.tsx

```diff
@@ -0,0 +1,45 @@
+import { FormControl } from '@mui/material';
+import TagManager from 'features/tags/components/TagManager';
+import { useState } from 'react';
+import { ZetkinTag } from 'utils/types/zetkin';
+
+import { COLUMN_TYPE, SelectedViewColumn } from '../types';
+
+interface PersonTagConfigProps {
+  onOutputConfigured: (columns: SelectedViewColumn[]) => void;
+}
+
+const PersonTagConfig = ({ onOutputConfigured }: PersonTagConfigProps) => {
+  const [selectedTags, setSelectedTags] = useState<ZetkinTag[]>([]);
+
+  const makeColumns = (tags: ZetkinTag[]) => {
+    return tags.map((tag) => ({
+      config: { tag_id: tag.id },
+      title: tag.title,
+      type: COLUMN_TYPE.PERSON_TAG,
+    }));
+  };
+
+  return (
+    <FormControl sx={{ width: 300 }}>
+      <TagManager
+        assignedTags={selectedTags}
+        groupTags={false}
+        onAssignTag={(tag) => {
+          const newAssignedTags = selectedTags.concat([tag] as ZetkinTag[]);
+          setSelectedTags(newAssignedTags);
+          const columns = makeColumns(newAssignedTags);
+          onOutputConfigured(columns);
+        }}
+        onUnassignTag={(tag) => {
+          const unassignedTags = selectedTags.filter((t) => t.id != tag.id);
+          setSelectedTags(unassignedTags);
+          const columns = makeColumns(unassignedTags);
+          onOutputConfigured(columns);
+        }}
```

Comment:
There's a lot of repetition here. I feel like if `setSelectedTags(…)` (lines 30 and 36) and `onOutputConfigured(…)` (lines 32 and 38) were moved into `makeColumns`, that would be a bit cleaner.

Doing so would of course change the purpose of `makeColumns`, so in that case we should probably also change the name to something like `onChangeTags`.

Reply:
I agree, that's a good solution to avoid the repetition!